account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1103 @@
+        # This is calculation is skipped for method_time != 'year'.
+        digits = self.env['decimal.precision'].precision_get('Account')
+        fy_residual_amount = self.depreciation_base
+        i_max = len(table) - 1
+        asset_sign = self.depreciation_base >= 0 and 1 or -1
+        line_dates = self._compute_line_dates(
+            table, depreciation_start_date, depreciation_stop_date)
+        for i, entry in enumerate(table):
+
+            if self.method_time == 'year':
+                year_amount = self._compute_year_amount(fy_residual_amount)
+                if self.method_period == 'year':
+                    period_amount = year_amount
+                elif self.method_period == 'quarter':
+                    period_amount = year_amount / 4
+                elif self.method_period == 'month':
+                    period_amount = year_amount / 12
+                if i == i_max:
+                    if self.method in ['linear-limit', 'degr-limit']:
+                        fy_amount = fy_residual_amount - self.salvage_value
+                    else:
+                        fy_amount = fy_residual_amount
+                else:
+                    firstyear = i == 0 and True or False
+                    fy_factor = self._get_fy_duration_factor(
+                        entry, firstyear)
+                    fy_amount = year_amount * fy_factor
+                if asset_sign * (fy_amount - fy_residual_amount) > 0:
+                    fy_amount = fy_residual_amount
+                period_amount = round(period_amount, digits)
+                fy_amount = round(fy_amount, digits)
+            else:
+                fy_amount = False
+                if self.method_time == 'number':
+                    number = self.method_number
+                elif self.method_time == 'end':
+                    number = len(line_dates)
+                period_amount = round(self.depreciation_base / number, digits)
+
+            entry.update({
+                'period_amount': period_amount,
+                'fy_amount': fy_amount,
+            })
+            if self.method_time == 'year':
+                fy_residual_amount -= fy_amount
+                if round(fy_residual_amount, digits) == 0:
+                    break
+        i_max = i
+        table = table[:i_max + 1]
+
+        # Step 2:
+        # Spread depreciation amount per fiscal year
+        # over the depreciation periods.
+        self._compute_depreciation_table_lines(
+            table, depreciation_start_date, depreciation_stop_date,
+            line_dates)
+
+        return table
+
+    def _get_depreciation_entry_name(self, seq):
+        """ use this method to customise the name of the accounting entry """
+        return (self.code or str(self.id)) + '/' + str(seq)
+
+    @api.multi
+    def _compute_entries(self, date_end, check_triggers=False):
+        # To DO : add ir_cron job calling this method to
```

Comment:
Same here, maybe instead of leaving TODOs on the code, create a ticket for this and remove the TODO or just remove the TODO if there is nothing to implement. TODOs won't be picked up by anyone https://en.wikipedia.org/wiki/Bystander_effect